src/FileMakerException.php

```diff
@@ -32,10 +32,10 @@ class FileMakerException extends \Exception
      * @param integer $code Error code.
      * @param null|\Exception $previous
      */
-    public function __construct($fm, $message = null, $code = null, $previous = null)
+    public function __construct($fm, $message = null, $code = -1, $previous = null)
     {
         $this->fm = $fm;
-        if ($code !== null) {
+        if (empty($message)) {
```

Comment:
The previous code is to make sure $code is NOT null.
The new code will process if $message IS empty.

Can you please explain more about your method here ?

Reply:
@anhyeuviolet My reasoning is that this will have the same result, since there are no cases so far where a `FileMakerException` is initialized with both `$message` and `$code` specified with a valid value. All exceptions are currently initialized in two variations:
- A string is passed as `$message` and no `$code` is specified (`$code` was in this case `null` previously and now `-1`)
- `null` is passed as `$message` and an integer is passed as `$code`

Furthermore I think that if in the future an exception is added that is initialized with both values specified, that a message originating closer to the source of the exception would likely be more specific and therefore preferred over one returned by `getErrorString`.